### PR TITLE
connect: create SI tokens in local scope

### LINKED ISSUE #8063: [question] Change consul SI tokens to be local?
Hi! I was testing nomad+consul connect in some test clusters. I tested to shut down the primary consul datacenter and saw that I wasn't able to run any connect task because the consul SI token can't be created due to it being global. Does it make sense/Can it be local so if the primary DC fails for some reason other datacenters can still work properly?

I will try to change the code here and set a `Local: true` to test it.

https://github.com/hashicorp/nomad/blob/master/nomad/consul.go#L220

Edit: to clarify, everyting thats running it's running ok, but can't deploy new allocations.

### PR DESCRIPTION
**Pull request by shoenig**
- Change consul SI tokens to be local
- docs: update documentation with connect acls changes

Closes #8063 #8068

### CODE REVIEW

File: .changelog/8068.txt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+```release-note:breaking-change
+connect: Consul Service Identity ACL tokens automatically generated for Connect services are now
+created as Local rather than Global tokens. Nomad clusters with Connect services making cross-Consul
+datacenter requests will need to ensure their Consul agents are configured with anonymous ACL tokens
+of sufficient node and service read permissions.
+```
+
+```release-note:breaking-change
+The minimum Consul version supported by Nomad's Connect integration is now Consul v1.8.0.
```
Comment (schmichael) [inline .changelog/8068.txt:9]: Do we need a `connect:` prefix here? I don't think so but just checking.

**Review by schmichael — APPROVED** [review]
Fantastic work!

**Comment by schmichael** [comment]
Make sure to link/close any related Github issues too.

File: .changelog/8068.txt (hunk 1 above)
Comment (shoenig) [reply .changelog/8068.txt]: Ah we do, good catch!

**Pull request merged**